src/timing.h

```diff
@@ -360,16 +376,21 @@ struct jl_timing_block_cpp_t {
     jl_timing_block_cpp_t& operator=(const jl_timing_block_cpp_t &) = delete;
     jl_timing_block_cpp_t& operator=(const jl_timing_block_cpp_t &&) = delete;
 };
-#define JL_TIMING(subsystem, event) jl_timing_block_cpp_t __timing_block(JL_TIMING_ ## subsystem, JL_TIMING_EVENT_ ## event); \
-    _TRACY_CTOR(__timing_block.block.tracy_ctx, #event, (jl_timing_enable_mask >> (JL_TIMING_ ## subsystem)) & 1)
+#define JL_COND_TIMING(enabled, subsystem, event) \
+    jl_timing_block_cpp_t JL_TIMING_BLOCK_NAME(subsystem, event)(enabled, JL_TIMING_ ## subsystem, JL_TIMING_EVENT_ ## event); \
+    if (enabled && jl_timing_enabled(JL_TIMING_ ## subsystem)) \
+        _TRACY_CTOR(JL_TIMING_BLOCK(subsystem, event)->tracy_ctx, #event, 1);
 #else
-#define JL_TIMING(subsystem, event) \
+#define JL_COND_TIMING(enabled, subsystem, event) \
     __attribute__((cleanup(_jl_timing_block_destroy))) \
-    jl_timing_block_t __timing_block; \
-    _jl_timing_block_ctor(&__timing_block, JL_TIMING_ ## subsystem, JL_TIMING_EVENT_ ## event); \
-    _TRACY_CTOR(__timing_block.tracy_ctx, #event, (jl_timing_enable_mask >> (JL_TIMING_ ## subsystem)) & 1)
+    jl_timing_block_t JL_TIMING_BLOCK_NAME(subsystem, event); \
+    _jl_timing_block_ctor(JL_TIMING_BLOCK(subsystem, event), enabled, JL_TIMING_ ## subsystem, JL_TIMING_EVENT_ ## event); \
+    if (enabled && jl_timing_enabled(JL_TIMING_ ## subsystem)) \
+        _TRACY_CTOR(JL_TIMING_BLOCK(subsystem, event)->tracy_ctx, #event, 1);
```

Comment:
```suggestion
    _TRACY_CTOR(JL_TIMING_BLOCK(subsystem, event)->tracy_ctx, #event, enabled && jl_timing_enabled(JL_TIMING_ ## subsystem));
```

This is a multi-statement macro. Also, internally it creates a local and stores its pointer in the timing block, so for lifetime reasons it shouldn't be put in an inner block

Probably best just to let `_TRACY_CTOR` handle the enable bit